### ISSUE TITLE
Question - Custom Metrics SupervisedEvaluator

### ISSUE THREAD

**Issue by cemde** [Project-MONAI/MONAI#1305]
I am trying to add custom metrics to `additional_metrics` of the `SupervisedEvaluator`.

The only Metric class in `MONAI` (`monai.metrics.DiceMetric`) contains an `__init__` and `__call__` method and does not inherit as far as I can see.

The documentation states one can use "more Ignite metrics that also attach to Ignite Engine." Looking up `ignite.metrics.Metric`, these Metric classes have `__init__`, `reset`, `update` and `compute` methods.

These two formats seem to contradict.

Following the MONAI tutorials and the ignite tutorials I wrote this Metric class:

``` python
class DiceCEMetric(ignite.metrics.Metric):
    """Dice and Xentropy loss"""

    def __init__(self, output_transform=lambda x: x):
        super().__init__()
        self.dice = monai.losses.DiceLoss(softmax=True)
        self.cross_entropy = nn.CrossEntropyLoss()
        self.metric = torch.Tensor([0])
        self.num_samples = torch.Tensor([0])
        super(DiceCEMetric, self).__init__(output_transform=output_transform)

    def reset(self):
        self.metric = torch.Tensor([0])
        self.num_samples = torch.Tensor([0])

    def update(self, output: Sequence[torch.Tensor]) -> None:
        y_pred, y_true = output
        dice = self.dice(y_pred, y_true)
        # CrossEntropyLoss target needs to have shape (B, D, H, W)
        # Target from pipeline has shape (B, 2, D, H, W)
        ce = self.cross_entropy(y_pred, y_true.argmax(dim=1).long())
        self.metric += dice + ce
        self.num_samples += y_true.shape[0]
    
    def compute(self):
        return self.metric / self.num_samples
```

However the `update` method only gets called once per epoch when I set a breakpoint inside this method. In addition, only one image is passed in (not even an entire batch). The `ignite` documentation states that `update` would be called with every batch. What am I doing wrong?

**Comment by Nic-Ma** [comment]
Hi @cemde ,

Thanks for your interest here.
The metrics **wrapper** for ignite is in this folder:
https://github.com/Project-MONAI/MONAI/tree/master/monai/handlers

Thanks.

**Comment by cemde** [comment]
Thanks for replying. May I ask what you mean with metric *wrapper*? I am using different handlers (Tensorboard, MetricLogger etc) to process the metrics. However, defining them when initiating some Evaluation or Training engine fails with the Ignite definition of a Metric class

**Comment by Nic-Ma** [comment]
The real metrics computation is under the `metrics` folder, the ignite format metrics in MONAI is under `handlers` folder and call the real metrics.

Thanks.

**Comment by cemde** [comment]
Thank you! I understand what you are saying now!
